src/SignalRServiceExtension/Config/SignalRFunctionsHostBuilderExtensions.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.Functions.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.IdentityModel.Tokens;
+using System;
+
+namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
+{
+    using SignalRConnectionInfoConfigureFunc = Func<AccessTokenResult, HttpRequest, SignalRConnectionDetail, SignalRConnectionDetail>;
+
+    /// <summary>
+    /// Extensions to add access token provider and SignalR connection configuration
+    /// </summary>
+    public static class SignalRFunctionsHostBuilderExtensions
+    {
+        /// <summary>
+        /// Adds default access token provider to validate access token.
+        /// </summary>
+        /// <param name="builder">Azure function host builder</param>
+        /// <param name="configureTokenValidationParameters">Token validation parameters</param>
+        /// <returns><see cref="IFunctionsHostBuilder"/>Azure function host builder</returns>
+        public static IFunctionsHostBuilder AddAuth(this IFunctionsHostBuilder builder, Action<TokenValidationParameters> configureTokenValidationParameters)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            if (configureTokenValidationParameters == null)
+            {
+                throw new ArgumentNullException(nameof(configureTokenValidationParameters));
+            }
+
+            builder.Services.AddSingleton<IAccessTokenProvider>(s =>
```

Comment:
>AddSingleton [](start = 29, length = 12)

TryAddSingleton? Could customer use their own IAccessTokenProvider?

Reply:
add tests for custom AccessTokenProvider?

---
In reply to: [381094953](https://github.com/Azure/azure-functions-signalrservice-extension/pull/90#discussion_r381094953) [](ancestors = 381094953)